--- build/tmp/openapi-templates/kotlin-client/build.gradle.mustache	2022-06-10 09:00:06.779453167 -0400
+++ build/resources/patches/kotlin-client/build.gradle.mustache	2022-06-10 14:30:44.078765868 -0400
@@ -2,7 +2,7 @@
 version '{{artifactVersion}}'
 
 wrapper {
-    gradleVersion = '6.8.3'
+    gradleVersion = '{{gradleVersion}}'
     distributionUrl = "https://services.gradle.org/distributions/gradle-$gradleVersion-all.zip"
 }
 
@@ -46,6 +46,12 @@
 apply plugin: 'kotlinx-serialization'
 {{/kotlinx_serialization}}
 
+files(
+    "local.build.gradle",
+    "local.build.gradle.kts",
+).filter({ it.exists() } as Spec)
+    .forEach({ apply from: it })
+
 repositories {
     maven { url "https://repo1.maven.org/maven2" }
 }
