--- build/tmp/openapi-templates/kotlin-client/build.gradle.mustache	2021-10-12 13:21:28.541946723 -0400
+++ build/resources/patches/kotlin-client/build.gradle.mustache	2021-10-12 13:33:25.602942118 -0400
@@ -43,6 +43,12 @@
 apply plugin: 'kotlinx-serialization'
 {{/kotlinx_serialization}}
 
+files(
+    "local.build.gradle",
+    "local.build.gradle.kts",
+).filter({ it.exists() } as Spec)
+    .forEach({ apply from: it })
+
 repositories {
     maven { url "https://repo1.maven.org/maven2" }
 }
