--- tmp/openapi-templates/kotlin-client/libraries/jvm-okhttp/api.mustache	2021-08-18 20:34:55.434736250 -0400
+++ resources/patches/kotlin-client/libraries/jvm-okhttp/api.mustache	2021-08-20 12:29:15.638660364 -0400
@@ -17,7 +17,10 @@
 import {{packageName}}.infrastructure.toMultiValue

 {{#operations}}
-{{#nonPublicApi}}internal {{/nonPublicApi}}class {{classname}}(basePath: kotlin.String = defaultBasePath) : ApiClient(basePath) {
+{{#nonPublicApi}}internal {{/nonPublicApi}}class {{classname}}(
+    basePath: kotlin.String = defaultBasePath,
+    accessToken: String? = null{{! Equisoft: Add optional accessToken}}
+) : ApiClient(basePath, accessToken) {
     companion object {
         @JvmStatic
         val defaultBasePath: String by lazy {
