--- tmp/openapi-templates/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiClient.kt.mustache	2021-08-18 20:34:55.434736250 -0400
+++ resources/patches/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiClient.kt.mustache	2021-08-20 12:29:15.648660205 -0400
@@ -56,7 +56,10 @@
 import org.threeten.bp.OffsetTime
 {{/threetenbp}}

-{{#nonPublicApi}}internal {{/nonPublicApi}}open class ApiClient(val baseUrl: String) {
+{{#nonPublicApi}}internal {{/nonPublicApi}}open class ApiClient(
+    val baseUrl: String,
+    accessToken: String? = null{{! Equisoft: Add optional accessToken}}
+) {
     {{#nonPublicApi}}internal {{/nonPublicApi}}companion object {
         protected const val ContentType = "Content-Type"
         protected const val Accept = "Accept"
@@ -81,6 +84,19 @@
         val builder: OkHttpClient.Builder = OkHttpClient.Builder()
     }

+    {{! Equisoft - Begin }}
+    private var internalAccessToken: String? = null
+    var accessToken: String?
+        get() = internalAccessToken ?: Companion.accessToken
+        set(value) {
+            internalAccessToken = value
+        }
+
+    init {
+        internalAccessToken = accessToken
+    }
+    {{! Equisoft - End }}
+
     /**
      * Guess Content-Type header from the given file (defaults to "application/octet-stream").
      *
