--- build/tmp/openapi-templates/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiClient.kt.mustache	2022-06-10 09:00:06.787453220 -0400
+++ build/resources/patches/kotlin-client/libraries/jvm-okhttp/infrastructure/ApiClient.kt.mustache	2022-06-10 14:36:57.429182548 -0400
@@ -77,7 +77,11 @@
 import com.squareup.moshi.adapter
 {{/moshi}}
 
-{{#nonPublicApi}}internal {{/nonPublicApi}}open class ApiClient(val baseUrl: String, val client: OkHttpClient = defaultClient) {
+{{#nonPublicApi}}internal {{/nonPublicApi}}open class ApiClient(
+    val baseUrl: String,
+    accessToken: String? = null, {{! Equisoft: Add optional accessToken}}
+    val client: OkHttpClient = defaultClient
+) {
     {{#nonPublicApi}}internal {{/nonPublicApi}}companion object {
         protected const val ContentType = "Content-Type"
         protected const val Accept = "Accept"
@@ -103,6 +107,19 @@
         val builder: OkHttpClient.Builder = OkHttpClient.Builder()
     }
 
+    {{! Equisoft - Begin }}
+    private var internalAccessToken: String? = null
+    var accessToken: String?
+        get() = internalAccessToken ?: Companion.accessToken
+        set(value) {
+            internalAccessToken = value
+        }
+
+    init {
+        internalAccessToken = accessToken
+    }
+    {{! Equisoft - End }}
+
     /**
      * Guess Content-Type header from the given file (defaults to "application/octet-stream").
      *
