--- tmp/openapi-templates/php/model_generic.mustache	2021-08-18 20:34:55.494735013 -0400
+++ resources/patches/php/model_generic.mustache	2021-08-20 11:49:59.582839952 -0400
@@ -1,6 +1,15 @@
 class {{classname}} {{#parentSchema}}extends {{{parent}}}{{/parentSchema}}{{^parentSchema}}implements ModelInterface, ArrayAccess, \JsonSerializable{{/parentSchema}}
 {
     public const DISCRIMINATOR = {{#discriminator}}'{{discriminatorName}}'{{/discriminator}}{{^discriminator}}null{{/discriminator}};
+{{! BEGIN Equisoft fix to handle mapped models }}
+{{#discriminator}}
+    public const DISCRIMINATOR_MAP = [
+    {{#discriminator.mappedModels}}
+        "{{mappingName}}" => "{{modelName}}",
+    {{/discriminator.mappedModels}}
+    ];
+{{/discriminator}}
+{{! END Equisoft fix to handle mapped models }}

     /**
       * The original name of the model.
