--- tmp/openapi-templates/php/ObjectSerializer.mustache	2021-08-18 20:34:55.491401747 -0400
+++ resources/patches/php/ObjectSerializer.mustache	2021-08-20 11:45:40.867260859 -0400
@@ -355,7 +355,12 @@
             // If a discriminator is defined and points to a valid subclass, use it.
             $discriminator = $class::DISCRIMINATOR;
             if (!empty($discriminator) && isset($data->{$discriminator}) && is_string($data->{$discriminator})) {
-                $subclass = '\{{invokerPackage}}\Model\\' . $data->{$discriminator};
+                {{! Equisoft fix to handle mapped models }}
+                if (isset($class::DISCRIMINATOR_MAP[$data->{$discriminator}])) {
+                    $subclass = '\{{invokerPackage}}\Model\\' . $class::DISCRIMINATOR_MAP[$data->{$discriminator}];
+                } else {
+                    $subclass = '\{{invokerPackage}}\Model\\' . $data->{$discriminator};
+                }
                 if (is_subclass_of($subclass, $class)) {
                     $class = $subclass;
                 }
