--- build/tmp/openapi-templates/php/ObjectSerializer.mustache	2022-06-10 09:00:06.879453834 -0400
+++ build/resources/patches/php/ObjectSerializer.mustache	2022-06-10 14:21:17.224563800 -0400
@@ -414,7 +414,7 @@
             // determine file name
             if (
                 is_array($httpHeaders)
-                && array_key_exists('Content-Disposition', $httpHeaders) 
+                && array_key_exists('Content-Disposition', $httpHeaders)
                 && preg_match('/inline; filename=[\'"]?([^\'"\s]+)[\'"]?$/i', $httpHeaders['Content-Disposition'], $match)
             ) {
                 $filename = Configuration::getDefaultConfiguration()->getTempFolderPath() . DIRECTORY_SEPARATOR . self::sanitizeFilename($match[1]);
@@ -449,7 +449,12 @@
             // If a discriminator is defined and points to a valid subclass, use it.
             $discriminator = $class::DISCRIMINATOR;
             if (!empty($discriminator) && isset($data->{$discriminator}) && is_string($data->{$discriminator})) {
-                $subclass = '\{{invokerPackage}}\Model\\' . $data->{$discriminator};
+                {{! Equisoft fix to handle mapped models }}
+                if (isset($class::DISCRIMINATOR_MAP[$data->{$discriminator}])) {
+                    $subclass = '\{{invokerPackage}}\Model\\' . $class::DISCRIMINATOR_MAP[$data->{$discriminator}];
+                } else {
+                    $subclass = '\{{invokerPackage}}\Model\\' . $data->{$discriminator};
+                }
                 if (is_subclass_of($subclass, $class)) {
                     $class = $subclass;
                 }
